src/main/java/org/truffleruby/extra/ffi/PointerNodes.java

```diff
@@ -156,23 +161,29 @@ public DynamicObject readString(DynamicObject pointer, int length) {
     public static abstract class PointerReadStringToNullPrimitiveNode extends PrimitiveArrayArgumentsNode {
 
         @Specialization(guards = "isNullPointer(pointer)")
-        public DynamicObject readNullPointer(DynamicObject pointer) {
-            return createString(RopeConstants.EMPTY_ASCII_8BIT_ROPE);
+        public DynamicObject readNullPointer(DynamicObject pointer,
+                                             @Cached("create()") AllocateObjectNode allocateObjectNode) {
+            return allocateObjectNode.allocate(coreLibrary().getStringClass(), Layouts.STRING.build(false, false, RopeConstants.EMPTY_ASCII_8BIT_ROPE));
```

Comment:
To clarify a bit, I wanted to remove that `createString` method and this was the lone remaining caller. I'll try to make creating strings from ropes cleaner is another pass.